miio/integrations/vacuum/mijia/pro2vacuum.py

```diff
@@ -0,0 +1,308 @@
+
+        [
+            {'did': 'battery', 'siid': 3, 'piid': 1, 'code': 0, 'value': 100},
+            {'did': 'error_code', 'siid': 2, 'piid': 2, 'code': 0, 'value': 0},
+            {'did': 'state', 'siid': 2, 'piid': 1, 'code': 0, 'value': 5},
+            {'did': 'fan_speed', 'siid': 2, 'piid': 6, 'code': 0, 'value': 1},
+            {'did': 'operating_mode', 'siid': 2, 'piid': 4, 'code': 0, 'value': 1},
+            {'did': 'mop_state', 'siid': 16, 'piid': 1, 'code': 0, 'value': 0},
+            {'did': 'water_level', 'siid': 2, 'piid': 5, 'code': 0, 'value': 2},
+            {'did': 'main_brush_life_level', 'siid': 14, 'piid': 1, 'code': 0, 'value': 99},
+            {'did': 'main_brush_time_left', 'siid': 14, 'piid': 2, 'code': 0, 'value': 17959}
+            {'did': 'side_brush_life_level', 'siid': 15, 'piid': 1, 'code': 0, 'value': 0 },
+            {'did': 'side_brush_time_left', 'siid': 15, 'piid': 2', 'code': 0, 'value': 0},
+            {'did': 'filter_life_level', 'siid': 15, 'piid': 2', 'code': 0, 'value': 0},
+            {'did': 'filter_time_left', 'siid': 15, 'piid': 2', 'code': 0, 'value': 0},
+            {'did': 'clean_area', 'siid': 9, 'piid': 1, 'code': 0, 'value': 0},
+            {'did': 'clean_time', 'siid': 9, 'piid': 2, 'code': 0, 'value': 0}
+            {'did': 'current_language', 'siid': 7, 'piid': 21, 'code': 0, 'value': 0}
+            ]"""
+        self.data = data
+
+    @property
+    def battery(self) -> int:
+        """Battery Level."""
+        return self.data["battery"]
+
+    @property
+    def error_code(self) -> int:
+        """Error code as returned by the device."""
+        return int(self.data["error_code"])
+
+    @property
+    def error(self) -> str:
+        """Human readable error description, see also :func:`error_code`."""
+        try:
+            return ERROR_CODES[self.error_code]
+        except KeyError:
+            return "Definition missing for error %s" % self.error_code
+
+    @property
+    def state(self) -> DeviceState:
+        """Vacuum Status."""
+        return DeviceState(self.data["state"])
+
+    @property
+    def fan_speed(self) -> SuctionState:
+        """Fan Speed."""
+        return SuctionState(self.data["fan_speed"])
+
+    @property
+    def operating_mode(self) -> SweepType:
+        """Operating Mode."""
+        return SweepType(self.data["operating_mode"])
+
+    @property
+    def mop_state(self) -> ClothState:
+        """Mop State."""
+        return ClothState(self.data["mop_state"])
+
+    @property
+    def water_level(self) -> WaterState:
+        """Water Level."""
+        return WaterState(self.data["water_level"])
+
+    @property
+    def main_brush_life_level(self) -> int:
+        """Main Brush Life Level(%)."""
+        return self.data["main_brush_life_level"]
+
+    @property
+    def main_brush_time_left(self) -> int:
+        """Main Brush Life Time Left(hours)."""
+        return self.data["main_brush_time_left"]
```

Comment:
Use `timedelta` for time durations, that will make it easy for downstreams to convert it as they wish to do.